src/main/kotlin/org/rust/ide/newProject/ui/AddUserTemplateDialog.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.newProject.ui
+
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.layout.panel
+import org.rust.ide.newProject.state.RsUserTemplate
+import org.rust.ide.newProject.state.RsUserTemplatesState
+import java.awt.Dimension
+import javax.swing.JComponent
+import javax.swing.event.DocumentEvent
+import javax.swing.event.DocumentListener
+
+class AddUserTemplateDialog : DialogWrapper(null) {
+    private val repoLinkField: JBTextField = JBTextField().apply {
+        preferredSize = Dimension(400, 0)
+        document.addDocumentListener(object : DocumentListener {
+            override fun insertUpdate(e: DocumentEvent?) {
+                if (nameField.text.isNotBlank()) return
+                if (listOf("https://", "git@").none { this@apply.text.startsWith(it) }) return
```

Comment:
Let's introduce a variable `val knownUrlPrefixes = setOf("http://", "https://", "git@")`

Reply:
Realized that cargo-generate does not support ssh git links, leaving only `http://` and `https://` for known prefixes

Reply:
Where should we leave `val knownUrlPrefixes`? In the `suggestName()` function, dialog class or in companion object?

Reply:
No matter IMO, but don't forget about `private` modifier if possible